framework/src/org/checkerframework/framework/util/QualifierPolymorphism.java

```diff
@@ -231,6 +237,13 @@ public void annotate(AnnotatedExecutableType functionalInterface, AnnotatedExecu
 
         List<AnnotatedTypeMirror> args = functionalInterface.getParameterTypes();
         List<AnnotatedTypeMirror> requiredArgs = memberReference.getParameterTypes();
+        if (args.size() == requiredArgs.size() + 1) {
+            // If the member reference is a reference to an instance method of an arbitrary
+            // object, then first parameter of the functional interface corresponds to the
+            // receiver of the member reference.
+            requiredArgs = new ArrayList<>(requiredArgs);
```

Comment:
Isn't this a rather inefficient way of doing this? This first copies requiredArgs into the new array, and then has to move everything over to add a first element.